CMakeLists.txt

```diff
@@ -1,56 +1,77 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.10)
 
 set(CMAKE_DISABLE_SOURCE_CHANGES ON)
 set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)
 
 project(tfs)
 
+add_subdirectory(src)
+add_executable(tfs ${tfs_SRC})
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-include(cotire)
+set_target_properties(tfs PROPERTIES CXX_STANDARD 11)
+set_target_properties(tfs PROPERTIES CXX_STANDARD_REQUIRED ON)
+
+if (${CMAKE_VERSION} VERSION_GREATER "3.16.0")
+    target_precompile_headers(tfs PUBLIC src/otpch.h)
+else ()
+    include(cotire)
+    set_target_properties(tfs PROPERTIES COTIRE_CXX_PREFIX_HEADER_INIT "src/otpch.h")
+    set_target_properties(tfs PROPERTIES COTIRE_ADD_UNITY_BUILD FALSE)
+    cotire(tfs)
+endif ()
 
-add_compile_options(-Wall -Werror -pipe -fvisibility=hidden)
+if (NOT WIN32)
+    add_compile_options(-Wall -Werror -pipe -fvisibility=hidden)
```

Comment:
I suggest removing `-pipe` here and let the user specify via build flags instead.

Reply:
I didn't add it, so it has only been "reorganized". We can remove it but I'm trying to limit changes here.

https://stackoverflow.com/questions/1512933/when-should-i-use-gccs-pipe-option